packages/run-protocol/src/vaultFactory/vault.js

```diff
@@ -617,20 +665,39 @@ export const makeVaultKit = (
     manager.reallocateReward(fee, vaultSeat, seat);
 
     trace(
-      'openLoan about to refreshLoanTracking: collateral',
+      'initVault about to refreshLoanTracking: collateral',
       getCollateralAmount(),
     );
     refreshLoanTracking(oldDebt, oldCollateral, runDebt, collateralAmount);
 
-    updateUiState();
-
-    return { notifier };
+    // TODO why isn't this complaining about use-before-define
+    updateOuter(innerVault);
+    return outerVault
   };
 
-  /** @type {Vault} */
-  const vault = Far('vault', {
+  const makeTransferInvitationHook = seat => {
+    assertVaultIsOpen();
+    throw 'TODO unimplemented';
+  }
+
+  const makeTransferInvitation = () => {
+    updateOuter(innerVault);
```

Comment:
I disallow all changes. That's clear and has many fewer possible testable states. It's also much clearer that access is revoked because reachability is revoked.  Hmm. This suggests that I should really null the innerVault pointer when the outer is revoked. Make the innerVault just unreachable via the outerVault. That might be more obviously reliable than the current test.